warehouse/query-core/src/main/java/datawave/next/SimpleQueryVisitor.java

```diff
@@ -0,0 +1,198 @@
+package datawave.next;
+
+import java.util.Set;
+
+import org.apache.commons.jexl3.parser.ASTAndNode;
+import org.apache.commons.jexl3.parser.ASTEQNode;
+import org.apache.commons.jexl3.parser.ASTERNode;
+import org.apache.commons.jexl3.parser.ASTFunctionNode;
+import org.apache.commons.jexl3.parser.ASTGENode;
+import org.apache.commons.jexl3.parser.ASTGTNode;
+import org.apache.commons.jexl3.parser.ASTJexlScript;
+import org.apache.commons.jexl3.parser.ASTLENode;
+import org.apache.commons.jexl3.parser.ASTLTNode;
+import org.apache.commons.jexl3.parser.ASTMethodNode;
+import org.apache.commons.jexl3.parser.ASTNENode;
+import org.apache.commons.jexl3.parser.ASTNRNode;
+import org.apache.commons.jexl3.parser.ASTNotNode;
+import org.apache.commons.jexl3.parser.JexlNode;
+
+import datawave.query.jexl.JexlASTHelper;
+import datawave.query.jexl.nodes.ExceededOr;
+import datawave.query.jexl.nodes.QueryPropertyMarker;
+import datawave.query.jexl.visitors.BaseVisitor;
+
+/**
+ * Validates that a query is composed of indexed terms with simple boolean logic. No index-only fields or functions.
+ */
+public class SimpleQueryVisitor extends BaseVisitor {
+
+    private final Set<String> indexedFields;
+    private final Set<String> indexOnlyFields;
+    private final Set<String> nonEventFields;
+
+    // assume noble intentions
+    private boolean valid = true;
+    private boolean negationsExist = false;
+    private boolean atLeastOneFieldIndexed = false;
+
+    public SimpleQueryVisitor(Set<String> indexedFields, Set<String> indexOnlyFields, Set<String> nonEventFields) {
+        this.indexedFields = indexedFields;
+        this.indexOnlyFields = indexOnlyFields;
+        this.nonEventFields = nonEventFields;
+    }
+
+    public boolean isValid() {
+        return this.valid && this.atLeastOneFieldIndexed && !negationsExist;
```

Comment:
is this condition okay with an or? Consider FRUIT:APPLE OR PROTIEN:STEAK OR CARB:FRENCHFRIES if one of these fields is indexed or index only this will return true.

Reply:
I'll add a todo here..